drivers/power/adp1050/iio_adp1050.h

```diff
@@ -0,0 +1,204 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef IIO_ADP1050_H
+#define IIO_ADP1050_H
+
+#include <stdbool.h>
+#include "iio.h"
+#include "adp1050.h"
+
+#define ADP1050_FLOAT_FREQ_INDEX		24
+#define ADP1050_FLOAT_FREQ(x)			((((x) - 0xF800) / 2))
+#define ADP1050_FLOAT_FREQ_ENUM(x)		((((x) * 2) + 0xF800) + 1)
+#define ADP1050_MODULATION_AVAIL_SIZE		8
+#define ADP1050_MODULATION_AVAIL_MASK(x)	NO_OS_GENMASK((((x) * 2) + 1), ((x) * 2))
+#define ADP1050_MODULATION_AVAIL_SELECT(x)	NO_OS_BIT(((x) * 2) + 1)
+
+static const char *const adp1050_enable_avail[2] = {
+	"Disabled", "Enabled"
+};
+
+static const char *const adp1050_freq_avail[] = {
+	"49KHz",
+	"59KHz",
+	"60KHz",
+	"65KHz",
+	"71KHz",
+	"78KHz",
+	"87KHz",
+	"104KHz",
+	"120KHz",
+	"130KHz",
+	"136KHz",
+	"142KHz",
+	"149KHz",
+	"184KHz",
+	"223KHz",
+	"250KHz",
+	"284KHz",
+	"329KHz",
+	"338KHz",
+	"347KHz",
+	"357KHz",
+	"379KHz",
+	"397KHz",
+	"403KHz",
+	"410KHz",
+	"97.5KHz",
+	"111.5KHz",
+	"156.5KHz",
+	"164.5KHz",
+	"173.5KHz",
+	"195.5KHz",
+	"201.5KHz",
+	"208.5KHz",
+	"215.5KHz",
+	"231.5KHz",
+	"240.5KHz",
+	"260.5KHz",
+	"271.5KHz",
+	"297.5KHz",
+	"312.5KHz",
+	"320.5KHz",
+	"367.5KHz",
+	"390.5KHz"
+};
+
+static const uint32_t adp1050_freq2_avail[] = {
+	49,
+	59,
+	60,
+	65,
+	71,
+	78,
+	87,
+	104,
+	120,
+	130,
+	136,
+	142,
+	149,
+	184,
+	223,
+	250,
+	284,
+	329,
+	338,
+	347,
+	357,
+	379,
+	397,
+	403,
+	410,
+	97,
+	111,
+	156,
+	164,
+	173,
+	195,
+	201,
+	208,
+	215,
+	231,
+	240,
+	260,
+	271,
+	297,
+	312,
+	320,
+	367,
+	390,
+};
+
+static const char *const adp1050_loop_avail[2] = {
+	[ADP1050_CLOSE_LOOP] = "Close Loop",
+	[ADP1050_OPEN_LOOP] = "Open Loop",
+};
+
+static const char *const adp1050_modulation_avail[] = {
+	[2] = "ADP1050_OUTA_SR1_FALLING_MOD_POSITIVE",
+	[3] = "ADP1050_OUTA_SR1_FALLING_MOD_NEGATIVE",
+	[8] = "ADP1050_OUTA_SR1_RISING_MOD_POSITIVE",
+	[12] = "ADP1050_OUTA_SR1_RISING_MOD_NEGATIVE",
+	[32] = "ADP1050_OUTB_SR2_FALLING_MOD_POSITIVE",
+	[48] = "ADP1050_OUTB_SR2_FALLING_MOD_NEGATIVE",
+	[128] = "ADP1050_OUTB_SR2_RISING_MOD_POSITIVE",
+	[192] = "ADP1050_OUTB_SR2_RISING_MOD_NEGATIVE",
+};
+
+enum adp1050_enable_type {
+	ADP1050_OUT_ENABLE,
+	ADP1050_FEEDFORWARD_ENABLE,
+	ADP1050_PULSE_ENABLE,
+	ADP1050_FREQ_SYNC_ENABLE
+};
+
+enum adp1050_vout_value_type {
+	ADP1050_VOUT_COMMAND_VALUE = ADP1050_VOUT_COMMAND,
+	ADP1050_VOUT_MARGIN_LOW_VALUE = ADP1050_VOUT_MARGIN_LOW,
+	ADP1050_VOUT_MARGIN_HIGH_VALUE = ADP1050_VOUT_MARGIN_HIGH,
+	ADP1050_VOUT_SCALE_MONITOR_VALUE = ADP1050_VOUT_SCALE_MONITOR,
+	ADP1050_VOUT_OFFSET_VALUE = ADP1050_VOUT_CAL_OFFSET,
+};
+
+enum adp1050_input_chan_type {
+	ADP1050_VIN_CHAN,
+	ADP1050_IIN_CHAN,
+	ADP1050_VOUT_CHAN,
+	ADP1050_TEMP_CHAN
+};
```

Comment:
do these definitions need to be in the header? it looks like they are only used in the c file so they should not be exported, and please use iio in the names

Reply:
They will be moved in the c file.